### PR TITLE
Openexr

### LINKED ISSUE #1802: Add OpenEXR
It would be great to have a package for OpenEXR.

cc @patricksnape

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Closes #1802. Closes #6913.

### CODE REVIEW

**Comment by aaronang** [comment]
I totally forgot to respond! Thank you for taking on this task.

**Comment by SylvainCorlay** [comment]
I just noticed that the `ilmbase` package also included `openexr` as part of the package! Also, it builds from the `openexr` source which vendors `ilmbase` instead of the original source of `ilmbase`... I noticed this as I was working on properly depending on ilmbase here...

Nothing depends on `ilmbase` at the moment besides this staged recipe.

I would find it really tempting to remove ilmbase from the gcc7 channel. -- cc @conda-forge/core

**Comment by npavlovikj** [comment]
@SylvainCorlay , the "build.sh" below worked fine for me on Linux. I believe setting the flags will fix the current Windows issue too:

#!/bin/bash

export INCLUDE_PATH="${PREFIX}/include/:${PREFIX}/include/OpenEXR/"
export LIBRARY_PATH="${PREFIX}/lib"
export LD_LIBRARY_PATH="${PREFIX}/lib"
export CPPFLAGS="$CPPFLAGS -I${PREFIX}/include -I${PREFIX}/include/OpenEXR"
export LDFLAGS="$LDFLAGS -L${PREFIX}/lib"

./configure --prefix=${PREFIX} --host="${HOST}" --build="${BUILD}" --with-ilmbase-prefix=${PREFIX} --enable-cxxstd=11
make -j ${CPU_COUNT}
make install

**Comment by SylvainCorlay** [comment]
Thanks! In the latest build, it seems that the remaining issue is that openexr requires a static build of ilmbase on windows.

**Comment by SylvainCorlay** [comment]
(Besides, linux and osx builds succeed.)

**Comment by npavlovikj** [comment]
Ah, I see. That is odd since the Windows build didn't use "OPENEXR_BUILD_STATIC=ON"... Have you tried explicitly setting OPENEXR_BUILD_STATIC=OFF here?

**Comment by SylvainCorlay** [comment]
@npavlovikj while this build is succesful, the windows build is not using the ilmbase package, but building its own. The cmake of openexr appears to be quite broken and not support that scenario well.

To not build against ilmbase, add

```
-DOPENEXR_BUILD_ILMBASE:BOOL=OFF ^
-DILMBASE_INCLUDE_DIR:PATH=%_LIBRARY_PREFIX%/include ^
```

in `bld.bat`. I have already spent a lot of time debugging the windows cmake...

**Comment by SylvainCorlay** [comment]
We could skip the windows build for now and get this in, and then work on the windows build with a PR.

File: recipes/openexr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    - test -e $PREFIX/lib/libIexMath$SHLIB_EXT         # [unix]
+    - test -e $PREFIX/lib/libIex$SHLIB_EXT             # [unix]
+    - test -e $PREFIX/lib/libIlmImf$SHLIB_EXT          # [unix]
+    - test -e $PREFIX/lib/libIlmImfUtil$SHLIB_EXT      # [unix]
+    - test -e $PREFIX/lib/libIlmThread$SHLIB_EXT       # [unix]
+    - test -e $PREFIX/lib/libImath$SHLIB_EXT           # [unix]
+
+about:
+  home: https://github.com/openexr/openexr
+  license: Modified BSD
```
Comment (ocefpaf) [inline recipes/openexr/meta.yaml]: We are trying to standardize this based on the SPDX. Can you change that BSD-3-Clause?

Comment (SylvainCorlay) [reply recipes/openexr/meta.yaml]: Sounds good. Doing it on the feedstock.

**Comment by jakirkham** [comment]
Thanks so much @aaronang, @SylvainCorlay, and @npavlovikj for working on this! 😄

File: recipes/openexr/meta.yaml (hunk 1 above)
Comment (SylvainCorlay) [reply recipes/openexr/meta.yaml]: Fixed.

**Pull request merged**